### DISCUSSION TITLE
Terminal Backend for Linux-only app

### DISCUSSION THREAD

**Discussion by lukeflo** [ratatui/ratatui#1537] [Q&A] [answered]
Hi,

I'm thinking about a little TUI app for Linux OS only (indeed only a specific distro).  I got some experience with Rust but I'm by far no full stack Rust dev.

For my last `ratatui` based project I used Crossterm backend which was already implemented in the template.

Now I'm wondering if there might be any performance/interaction advantage using Termion backend over Crossterm when programming an app only for Linux.

**Comment by joshka** [comment, marked as answer]
As a general rule I prefer using common tools over more unique ones, for a lower complexity budget and maintenance burden, so crossterm would be my general choice for any app.

That said:
Your best bet is to give it a try. I doubt it would be significant. (I'd measure it if it mattered)
Try converting the colors_rgb to termion and seeing what fps you get for the output side.
The input (events) side is more difficult as it's hard to measure input latency in a meaningful way.
If you need to test it in real world use, consider using an old raspberry pi or a CPU limited VM to artificially make everything slower in a way that highlights any perf issue.

**Reply by lukeflo** [reply]
It shouldn't matter much for my planned app in particular since its not a heavy thing regarding workload.

First of all, I was interested if a OS specific thing like Termion has some serious advantages over a program which focusses on supporting a width range of systems and, therefore, is very likely to have a larger codebase like I assume for Crossterm.

Taking your arguments into account, I guess I stick with Crossterm for now, since I'm already a little bit familiar with it. Maybe I'll try Termion some day in the future if I got too much time.

Thanks for your advice and keep up the great work!
